.github/workflows/test_dbt_models.yaml

```diff
@@ -73,3 +75,18 @@ jobs:
 
             Link to failing workflow:
             https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+      - name: Send data test failure notification
+        run: |
+          failure_notifications_file="$PROJECT_DIR/qc_test_results/failure_notifications.json"
+          if [ -f "$failure_notifications_file" ]; then
+            echo "Sending failure notifications"
+            bash <( \
+              jq -r \
+              '.[] | "./.github/actions/publish_sns_topic/publish.sh \(.topic_arn|@sh) \(.subject|@sh) \(.body|@sh)"' \
+              "$failure_notifications_file" \
```

Comment:
This unholy jq command was the simplest way I could think of parsing failures by topic and queuing up each one to send, but I'm open to ideas that will be more understandable to laypeople!

Reply:
This is definitely a little bit cursed. What about doing it in Python? Also, let's add some conditionals on this (like those on L68) so that we don't accidentally spam folks if this workflow is triggered a bunch.

Reply:
Great ideas on both fronts! I gated this step behind a conditional in e5a615e24a91e436fb06014e4a822031353c703a and factored the esoteric jq call into a Python script in 8fbac8997901e26967054274df16adc9103704e0. Note that I decided to implement the gate via an input variable, so we'll have to update the Spark job in order to enable it. I think this is a feature rather than a bug, however, since it means we can merge this change without enabling it immediately.